### DISCUSSION TITLE
how to get it to work on computers that don't have it installed

### DISCUSSION THREAD

**Discussion by meltremb** [codam-coding-college/MLX42#85] [Q&A] [answered]
I'm at Quebec42, we're allowed to use the codam MLX for the project but we need to figure it out ourselves. we only have the default mlx on the computers, not even glfw. my so long works perfectly on all computers that have the codam mlx installed, but when i get corrected i may get people who don't have it. how do i get my so long to work on other computers?

**Comment by W2Wizard** [comment, marked as answer]
So if you have no access at all to either GLFW, CMake or if your own Sysadmin refuses or is unable to install any of it. Here are the steps to install all the necessary dependencies:

> Note: If you're on MacOS your sysadmin can also request the necessary Managed Software Center files for Munki regarding CMake and GLFW from our sysadmin at Codam (@FreekBes).

## I have no CMake!

1. Install [42Homebrew](https://github.com/kube/42homebrew)
2. Run `brew install cmake`, wait a bit. Brew is a slow mf :P
3. Restart your shell environment: `exec zsh` for example.

> Note: Once #84 has been merged CMake will be able to detect the brew installation of glfw so you can skip the second step entirely and just run `brew install glfw` (it will also fetch cmake with that)

## I have no GLFW!

1. Simply run `cmake -B build && cmake --build build -j4`. This will fetch GLFW for you if it can't find it.

2. For your makefile, go to `build/_deps/glfw-build/src` in there you will find the static library `libglfw3.a`. You need to add it to your compilation: `gcc main.c build/libmlx42.a build/_deps/glfw-build/src/libglfw.a ...`

3. Profit!

**Reply by W2Wizard** [reply]
In regards to CMake, thats the bare minimum. If you can't install that at the very least theres not much I can do, I mean how else will you build if you dont have the build system?

What you could do is then make your makefile check if its present, install homebrew for them, install cmake and then continue. Sounds a bit jank and like weird but that way you can just type `make` and wait for a while so it all works.
